tests/interfaces/test_qnode_autograd.py

```diff
@@ -639,19 +639,21 @@ def circuit2(data, weights):
             qml.templates.StronglyEntanglingLayers(weights, wires=[0, 1])
             return qml.expval(qml.PauliX(0))
 
-        def cost(weights):
-            w1, w2 = weights
+        def cost(w1, w2):
             c1 = circuit1(w1)
             c2 = circuit2(c1, w2)
             return np.sum(c2) ** 2
 
-        w1 = np.random.random(qml.templates.StronglyEntanglingLayers.shape(n_layers=3, n_wires=2))
-        w2 = np.random.random(qml.templates.StronglyEntanglingLayers.shape(n_layers=4, n_wires=2))
+        w1 = qml.templates.StronglyEntanglingLayers.shape(n_layers=3, n_wires=2)
+        w2 = qml.templates.StronglyEntanglingLayers.shape(n_layers=4, n_wires=2)
 
-        weights = [w1, w2]
+        weights = [
+            np.random.random(w1, requires_grad=True),
+            np.random.random(w2, requires_grad=True),
+        ]
 
         grad_fn = qml.grad(cost)
-        res = grad_fn(weights)
+        res = grad_fn(*weights)

```

Comment:
out of curiosity, how come this had to change?

Is it because `list[tensor]` wasn't registering as differentiable?

Reply:
Yes exactly :) and since the arguments are of different shape they cannot be combined into a single array.